pypeit/core/flexure.py

```diff
@@ -302,134 +304,124 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
-                         corr_cen=[], spec_file=sky_file, smooth=[],
-                         arx_spec=[], sky_spec=[])
+        flex_dict = dict(polyfit=None, shift=None, subpix=None, corr=None,
+                         corr_cen=None, spec_file=sky_file, smooth=None,
+                         arx_spec=None, sky_spec=None)
         # If no objects on this slit append an empty dictionary
         if islit not in gdslits:
             flex_list.append(flex_dict.copy())
             continue
-        for ss, specobj in enumerate(this_specobjs):
-            if specobj is None:
-                continue
-            if specobj['BOX_WAVE'] is None: #len(specobj._data.keys()) == 1:  # Nothing extracted; only the trace exists
-                continue
-            msgs.info("Working on flexure for object # {:d}".format(specobj.OBJID) + "in slit # {:d}".format(islit))
-            # Using boxcar
-            if method in ['boxcar', 'slitcen']:
-                sky_wave = specobj.BOX_WAVE #.to('AA').value
-                sky_flux = specobj.BOX_COUNTS_SKY
-            else:
-                msgs.error("Not ready for this flexure method: {}".format(method))
-
-            # Generate 1D spectrum for object
-            obj_sky = xspectrum1d.XSpectrum1D.from_tuple((sky_wave, sky_flux))
-
-            # Calculate the shift
-            fdict = spec_flex_shift(obj_sky, sky_spectrum, sky_lines, mxshft=mxshft)
-            punt = False
-            if fdict is None:
-                msgs.warn("Flexure shift calculation failed for this spectrum.")
-                if sv_fdict is not None:
-                    msgs.warn("Will used saved estimate from a previous slit/object")
-                    fdict = copy.deepcopy(sv_fdict)
-                else:
-                    # One does not exist yet
-                    # Save it for later
-                    return_later_sobjs.append([islit, ss])
-                    punt = True
-            else:
-                sv_fdict = copy.deepcopy(fdict)
-
-            # Punt?
-            if punt:
-                break
 
-            # Interpolate
-            new_sky = specobj.flexure_interp(sky_wave, fdict)
-            # Update dict
-            for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
-                flex_dict[key].append(fdict[key])
-            flex_dict['sky_spec'].append(new_sky)
+        slit_spat = slits.spat_id[islit]
+        mask = (slitmask == slit_spat) & gpm
 
-        flex_list.append(flex_dict.copy())
+        # Fill in the boxcar extraction tags
+        box_denom = moment1d(waveimg * mask > 0.0, traces[:, islit], 2, row=spec_vec)[0]
+        wghts = (box_denom + (box_denom == 0.0))
+        sky_flux = moment1d(skyimg * mask, traces[:, islit], 2, row=spec_vec)[0] / wghts
+        # Denom is computed in case the trace goes off the edge of the image
+        sky_wave = moment1d(waveimg * mask, traces[:, islit], 2, row=spec_vec)[0] / wghts
```

Comment:
Let's chat. I don't see why you are extracting a sky spectrum here. As I said in Slack, you have a bspline model for the sky as a function of the spectral pixel position, so you can just evaluate that model on a grid of spectral points, and this can be what you cross-correlate with the archival sky. There is no reason to extract a sky spectrum from the skyimg model.